### PR TITLE
ES6: Convert discussion/loader

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

ES6: Converts `discussion/loader.js`

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/discussion/loader.js [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+            }
+        );
+
+        if (config.page.section === 'crosswords') {
+            const $timestampsLabel = $('.js-timestamps');
+            const updateLabelText = prefValue => {
+                $timestampsLabel.text(prefValue ? 'Relative' : 'Absolute');
+            };
+
+            updateLabelText(undefined);
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/loader.js]: Unnecessary param 😸

File: static/src/javascripts/projects/common/modules/discussion/loader.js [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+            this.initState();
+            this.renderCommentBar();
+
+            if (this.user) {
+                this.comments.addUser(this.user);
+
+                const userPageSize = userPrefs.get('discussion.pagesize');
+                let pageSize = defaultPagesize;
+
+                if (isNumber(userPageSize)) {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/loader.js]: Can we kill this lodash dep?

```js
typeof userPageSize === 'number'
```

File: static/src/javascripts/projects/common/modules/discussion/loader.js [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+        // When the pagesize is 'All', the full page is not yet loaded, so load the comments.
+        if (this.comments.isAllPageSizeActive()) {
+            this.loadComments();
+        } else {
+            this.removeState('truncated');
+        }
+    }
+
+    renderCommentBox(elem: HTMLElement): void {
+        return new CommentBox({
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/loader.js]: I don't think we need a return statement here, as the return value will always be `undefined`

File: static/src/javascripts/projects/common/modules/discussion/loader.js [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+    renderCommentBar(): void {
+        if (this.isCommentable()) {
+            this.renderCommentBox(qwery('.js-discussion-comment-box--top')[0]);
+            this.renderCommentBox(
+                qwery('.js-discussion-comment-box--bottom')[0]
+            );
+        }
+    }
+
+    renderCommentCount() {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/loader.js]: Missing return type 🌺

File: static/src/javascripts/projects/common/modules/discussion/loader.js [hunk 5]
```diff
@@ -0,0 +1,589 @@
+// @flow
+
+import bean from 'bean';
+import bonzo from 'bonzo';
+import qwery from 'qwery';
+import $ from 'lib/$';
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/loader.js]: Ah, the holy quadrilogy of redundancy 😏

**Review by SiAdcock — APPROVED** [review]
Great conversion, man, this is an epic module!

**Comment by gustavpursche** [comment]
@nicl Thanks for reverting!

**Pull request merged**